Run blocking update

bootstraponline opened the pull request.

Commit:
Change root project name

* ignore local.properites

Commit:
Make expected runBlocking signature same as kotlinx.coroutines.runBlocking

jan-goral committed:
Move src to common module

jan-goral committed:
Add android app module

jan-goral committed:
Add healthCheck for android device

jan-goral committed:
Remove unnecessary code

jan-goral committed:
Apply some build error fixes

* Move commonTest to jvmTest for prevent kotlin test conflict with android JUnit
* Fix common building errors
* Simplify common kotlin targets declaration

jan-goral committed:
Generate podspec for common module

jan-goral committed:
Add fresh ios project

jan-goral committed:
Add Podfile with linked common.podspec to ios and install it

jan-goral committed:
Add healthCheck test for ios

bootstraponline commented on ios/runblockingTests/runblockingTests.swift:
```diff
@@ -0,0 +1,17 @@
+//  runblockingTests
+//
+//  Created by Janek Góral on 31/01/2020.
+//  Copyright © 2020 gogoapps. All rights reserved.
```
Let's remove the default comments from Xcode.

jan-goral replied:
Ok, I'll do it.

bootstraponline commented on common/common.podspec:
```diff
@@ -0,0 +1,44 @@
+            :script => <<-SCRIPT
+                set -ev
+                REPO_ROOT="$PODS_TARGET_SRCROOT"
+                "$REPO_ROOT/../gradlew" -p "$REPO_ROOT" :common:syncFramework \
```
We intentionally don't use Cocoapods because it's slow and introduces a Ruby dependency. Here's an example of accomplishing the same task with an Xcode shell script:

- https://github.com/russhwolf/multiplatform-hello/blob/d880b3579895c1cf11fb291624113829426affe7/ios/Multiplatform%20Hello.xcodeproj/project.pbxproj#L301

jan-goral replied:
Thanks for review, I'll implement fix ASAP.

bootstraponline commented:
@jan-gogo I left some comments. Nice work on the project so far. The example apps are helpful for testing.

jan-goral committed:
Add iOS dispatcher for testing purpose
Freezes anyway as described here https://github.com/ktorio/ktor/issues/678#issuecomment-433435243

* Upgrade kotlin coroutines to 1.3.3
* Remove default comments from iOS sources
* Fix package name clash between common and android

jan-goral committed:
Remove cocoapods for performance reason

jan-goral committed:
Replace build.gradle with build.gradle.kts

jan-goral committed:
Add common framework to ios project manually

* rename common to CommonRunBlocking (due to some errors with accessing native code)
* add simplified packForXcode script

bootstraponline commented on common/build.gradle.kts:
```diff
@@ -97,3 +97,20 @@ android {
+    into(targetDir)
+}
+
+tasks.getByName("build").dependsOn(packForXcode)
```
![image](https://user-images.githubusercontent.com/1173057/73763075-cb655800-4725-11ea-9cdd-c5581c89adce.png)

Small comment, it's good to end with a newline. 🙂

jan-goral replied:
Right. Small thing, I forgot about that. Fixed

jan-goral committed:
Add missing newlines
